{
  Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler(){
    public void uncaughtException(    Thread thread,    Throwable thrown){
      try {
        handleUncaughtException(thrown);
      }
 catch (      Error err) {
        LOG.error("Received error in main thread.. terminating server...",err);
        Runtime.getRuntime().exit(-2);
      }
    }
  }
);
}
